src/database/database.go

```diff
@@ -0,0 +1,186 @@
+package database
+
+import (
+	"context"
+	"database/sql"
+	"fmt"
+	"github.com/appuio/appuio-cloud-reporting/pkg/db"
+	"github.com/jmoiron/sqlx"
+	"github.com/vshn/cloudscale-metrics-collector/pkg/categoriesmodel"
+	"github.com/vshn/cloudscale-metrics-collector/pkg/datetimesmodel"
+	"github.com/vshn/cloudscale-metrics-collector/pkg/discountsmodel"
+	"github.com/vshn/cloudscale-metrics-collector/pkg/factsmodel"
+	"github.com/vshn/cloudscale-metrics-collector/pkg/productsmodel"
+	"github.com/vshn/cloudscale-metrics-collector/pkg/queriesmodel"
+	"github.com/vshn/cloudscale-metrics-collector/pkg/tenantsmodel"
+	"os"
+	"strings"
+	"time"
+
+	ctrl "sigs.k8s.io/controller-runtime"
+)
+
+const (
+	dbUrlEnvVariable   = "ACR_DB_URL"
+	sourceQueryStorage = "object-storage-storage"
+	provider           = "exoscale"
+	queryAndZone       = sourceQueryStorage + ":" + provider
+	defaultUnit        = "GBDay"
+)
+
+var (
+	productRow = db.Product{
```

Comment:
Variable naming: From the perspective of the application this is not a "row". The fact that it is a row in the DB should not concern the application; for the application it's just a persisted object or struct.